packages/@sanity/desk-tool/src/panes/document/documentPanel/documentViews/FormView.tsx

```diff
@@ -60,15 +61,23 @@ export function FormView(props: FormViewProps) {
     patchChannelRef.current = FormBuilder.createPatchChannel()
   }
 
+  const resolvedReadOnly = useResolveConditionalProperty({
+    document: value as SanityDocument,
+    value: undefined,
+    checkProperty: documentSchema.readOnly,
+    checkPropertyKey: 'readOnly',
+  })
+
   const readOnly = useMemo(() => {
     return (
       !ready ||
       rev !== null ||
       !permission.granted ||
       !isActionEnabled(documentSchema, 'update') ||
-      (isNonExistent && !isActionEnabled(documentSchema, 'create'))
+      (isNonExistent && !isActionEnabled(documentSchema, 'create')) ||
+      resolvedReadOnly
```

Comment:
I don't see why we shouldn't, but are we sure we want to add `readOnly` support on the document level since it wasn't supported before?

Reply:
It was supported, it just didn't show visually that they become readOnly, apparently

Reply:
Ah, you're right! We haven't documented it here either https://www.sanity.io/docs/document-type

Reply:
yeah! will let martin know